### PR TITLE
DCD 1220: allow bastion host deletion

### PR DESCRIPTION
**Pull request by bpar476**
# Summary

The bastion host had a `DeletionPolicy: Retain` which caused the entire VPC stack to fail to delete. The bastion EC2 instance wouldn't delete, so the security group wouldn't delete and nor could the VPC due to the retained EC2 instance depending on these resources.

We originally set the policy to retain so that when the stack is updated the EC2 instance is preserved and does not acquire a new IP address. However, this can be satisfied with the `UpdateReplacePolicy`. 

I also removed the `.DS_Store` because that shouldn't be there.

## Testing Process

1. At first I deployed an entire `jira-dc-with-vpc` stack and verified the VPC wouldn't be deleted. 
1. Next I deployed just the bastion stack and tested that it couldn't be deleted. 
1. Then I removed the `DeletionPolicy`, deployed a Bastion stack and deleted it. 

I tried to update the `DELETE_FAILED` stack with a bastion stack that has no `DeletionPolicy` but you can't update a stack when it is in `DELETE_FAILED` state.

### CODE REVIEW

**Comment by bpar476** [comment]
Green build https://server-syd-bamboo.internal.atlassian.com/browse/DCD-AWSSERVICES26-3

File: templates/quickstart-bastion-for-atlassian-services.yaml [hunk 1]
```diff
@@ -50,7 +50,7 @@ Parameters:
 Resources:
   Bastion:
     Type: AWS::EC2::Instance
-    DeletionPolicy: Retain
+    DeletionPolicy: Delete
     UpdateReplacePolicy: Retain
```
Comment (nanux) [inline templates/quickstart-bastion-for-atlassian-services.yaml:54]: Ok, so when there are changes in the parent stack - e.g. number of nodes, it won't try to replace the bastion host.
How does it behave if you change `BastionHostRequired` in the product-with-vpc stack from `true` to `false`? Does it remove or retain the EC2 instance?

**Review by nanux — APPROVED** [review]
Just one small clarification question - see the comment.

After merging this, we also need to bump the submodule commit in the product repos.

File: templates/quickstart-bastion-for-atlassian-services.yaml (hunk 1 above)
Comment (bpar476) [reply templates/quickstart-bastion-for-atlassian-services.yaml:54]: Good question. I can test that.

**Comment by bpar476** [comment]
> Just one small clarification question - see the comment.
> 
> After merging this, we also need to bump the submodule commit in the product repos.

Do we need to bump submodule commit in product repos? I believe taskcat will do that automatically once merged into the upstream and passing CI

**Comment by bpar476** [comment]
When updating the stack so that "Provision bastion host" is false, the bastion stack is deleted which causes the bastion instance to be deleted (due to the deletion policy)
image

**Comment by bpar476** [comment]
image

**Pull request merged**